rafthttp/stream.go

```diff
@@ -172,6 +174,10 @@ func (cw *streamWriter) resetCloser() {
 	defer cw.mu.Unlock()
 	if cw.working {
 		cw.closer.Close()
+		if len(cw.msgc) > 0 {
+			cw.r.ReportUnreachable(uint64(cw.id))
+			drainMessageChan(cw.msgc)
```

Comment:
if so, it needs to introduce a lock on msgc. i don't really like a lock on message channel.